ldap/servers/slapd/back-ldbm/ldbm_search.c

```diff
@@ -1515,6 +1517,17 @@ ldbm_back_next_search_entry(Slapi_PBlock *pb)
 
     /* Find the next candidate entry and return it. */
     while (1) {
+        if (li->li_dblock_monitoring && li->li_dblock_threshold_reached) {
```

Comment:
This isn't thread safe - these are int values and because they are traversing thread boundaries, search threads have no guarantee that they will see these changes from the lock monitor thread. These should be changed to be atomics (ordering relaxed) to ensure the values are syncing.